File: cpp/trees/bfs.cpp
#include <iostream>
#include <queue>

class TreeNode
{
public:
	double data;
	TreeNode *left;
	TreeNode *right;

	TreeNode(double data) { this->data = data; this->left = nullptr; this->right = nullptr;}
};

void breadth_first_search(TreeNode *root)
{
	std::queue<TreeNode *> queue;
	if (root != nullptr){
		queue.push(root);
		while (!queue.empty()) {
			TreeNode *node = queue.front();
			queue.pop();
			std::cout << node->data << std::endl;
			if (node->left != nullptr) { queue.push(node->left);}
			if (node->right != nullptr) { queue.push(node->right);}
		}
	}
}


int main() {
	return 0;
}

File: lc/py/bfs.py
def search(root):
  queue = []
  root.visited = true;
  visit(root)
  queue.append(root)

  while(len(queue) > 0):
    # Remove from the front of the queue
    r = queue.pop(0) 
    for n in r.adjacent:
      if not n.visited:
        visit(n)
        n.visited = True
        queue.append(n)



